test/helpers/outputs/fakeadapter/fake_adapter.go

```diff
@@ -0,0 +1,25 @@
+package fakeadapter
```

Comment:
Consider following the kubernetes pattern for client where they use a package like:

test/helpers/outputs/adapter/fake
internal/generator/vector/helpers/adapter/fake
internal/generator/vector/helpers/adapters/outputs/fake

Doing so would also mean you could drop "Fake" from method and type names